tests/workflows/test_publish_rolling_release.py

```
@@ -0,0 +1,383 @@
+        for index, command in enumerate(run.commands)
+        if "git/refs/tags/rolling" in command
+    ]
+    assert len(tag_moves) == 1, "the tag must move exactly once"
+    uploads = [
+        index
+        for index, command in enumerate(run.commands)
+        if "release upload" in command
+    ]
+    assert uploads and max(uploads) < tag_moves[0], (
+        "the tag moved while assets were still being uploaded"
+    )
```

Ruff (0.16.3)

[warning] Assertion should be broken down into multiple parts

(PT018)